### PR TITLE
Fix problem creating MaskedColumn from list of MaskedArray

### LINKED ISSUE #8816: Table column with maked data drops mask
[EDIT]: See https://github.com/astropy/astropy/issues/8816#issuecomment-500598908 for a more direct example of the bug.

-------

```
import numpy as np
from astropy.table import Table
data = np.ma.masked_greater(np.arange(5), 3)
tab = Table({'data': [data]})
mtab = Table({'data': [data]}, masked=True)
data, tab['data'][0], mtab['data'][0]
```
gives:
```
(masked_array(data=[0, 1, 2, 3, --],
              mask=[False, False, False, False,  True],
        fill_value=999999),
 array([0, 1, 2, 3, 4]),
 masked_BaseColumn(data=[0, 1, 2, 3, 4],
                   mask=[False, False, False, False, False],
             fill_value=999999))
```
So neither the table, nor the masked table respects the mask of the input data and silently converts the array to use the "data behind the mask".

CC:@taldcroft

### PR DESCRIPTION
**Pull request by taldcroft**
Fixes #8816

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -298,6 +298,12 @@ astropy.stats
 astropy.table
 ^^^^^^^^^^^^^
 
+- Fix the conversion of bytes values to Python ``str`` with ``Table.tolist``.
```
Comment (taldcroft) [inline CHANGES.rst]: Argh, I have no idea how this crept in.

Comment (bsipocz) [reply CHANGES.rst]: you didn't start from a brand new master branch. We've merged in yesterday a changelog post release cleanup.

Comment (taldcroft) [reply CHANGES.rst]: I did rebase on master up through 368dc06d1.  In any case that fix (#8739) is for 3.2 not 3.2.1 so I'm removing it from my branch here.

File: astropy/table/column.py [hunk 2]
```diff
@@ -1191,12 +1191,21 @@ def __new__(cls, data=None, name=None, mask=None, fill_value=None,
...
-            # is extremely slow (~3 seconds for 1e7 elements), while mask=False
-            # gets quickly broadcast to the expected bool array of False.
-            mask = getattr(data, 'mask', False)
-            if mask is not False:
-                mask = np.array(mask, copy=copy)
+            # If mask is None then we need to determine the mask (if any) from the data.
+            # The naive method is looking for a mask attribute on data, but this can fail,
+            # see #8816.  Instead use ``MaskedArray`` to do the work.
+            data_ma = ma.MaskedArray(data)
+            mask = data_ma.mask.copy() if copy else data_ma.mask
```
Comment (mhvk) [inline astropy/table/column.py]: Maybe deal with the `np.bool_` case, which is just `nomask` here as well (was thinking of falling through but we don't want the `deepcopy`):
```
mask = data_ma.mask
if mask is np.ma.nomask:
    mask = False
elif copy:
    mask = mask.copy()
```

Comment (mhvk) [reply astropy/table/column.py]: Actually, could make first line:
```
mask = ma.MaskedArray(data).mask
```

Comment (taldcroft) [reply astropy/table/column.py]: Awesome idea, done.  (And I came up with the `mask = ma.MaskedArray(data).mask` independently once I saw the code!)

**Pull request merged**